src/main/java/org/launchcode/water_garden_tour/models/garden/GardenData.java

```diff
@@ -0,0 +1,102 @@
+package org.launchcode.water_garden_tour.models.garden;
+
+import org.springframework.security.core.parameters.P;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GardenData {
+
+    public static List<Garden> gardenSearchFeaturesFiltered(String searchValueRaw, Integer[] featureIds, Iterable<Feature> allFeatures, List<Garden> allGardens) {
+        String searchValue = searchValueRaw.toLowerCase();
+        List<Garden> results = new ArrayList<>();
+        Feature curFeature;
+
+        for (Garden garden : allGardens) {
+            boolean hasAllFeatures = true;
+
+            for (Integer featureId : featureIds) {
+                curFeature = getFeatureFromId(featureId, allFeatures);
+
+                if (garden.getFeatures().contains(curFeature)) {
+                } else {
+                    hasAllFeatures = false;
+                }
```

Comment:
you could simply not what is inside of the if statement and then do away with the else case here:
`if (garden.getFeatures().contains(curFeature)) { hasAllFeatures = false } `

Reply:
Good call! Update made.